test/hotspot/jtreg/compiler/c2/irTests/ModDNodeTests.java

```diff
@@ -122,21 +128,31 @@ public double veryNotConstant(double x, double y) {
 
     @Test
     @IR(failOn = {"drem"}, phase = CompilePhase.BEFORE_MATCHING)
+    @IR(failOn = IRNode.MOD_D, phase = CompilePhase.ITER_GVN1)
+    @IR(counts = {IRNode.MOD_D, "1"}, phase = CompilePhase.AFTER_PARSING)
     public void unusedResult(double x, double y) {
         double unused = x % y;
     }
 
     @Test
     @IR(failOn = {"drem"}, phase = CompilePhase.BEFORE_MATCHING)
+    @IR(failOn = IRNode.MOD_D, phase = CompilePhase.ITER_GVN1)
+    @IR(counts = {IRNode.MOD_D, "1"}, phase = CompilePhase.AFTER_PARSING)
     public void repeatedlyUnused(double x, double y) {
         double unused = 1.d;
         for (int i = 0; i < 100_000; i++) {
             unused = x % y;
         }
     }
 
+    // The difference between unusedResultAfterLoopOpt1 and unusedResultAfterLoopOpt2
+    // is that they exercise a slightly different reason why the node is being removed,
+    // and thus a different execution path. In unusedResultAfterLoopOpt1 the modulo is
+    // used in the traps of the parse predicate. In unusedResultAfterLoopOpt2, it is not.
     @Test
     @IR(failOn = {"drem"}, phase = CompilePhase.BEFORE_MATCHING)
```

Comment:
This is not required now anymore since you do the matching on the node directly instead of the expanded call. Same for the other rules below.

Reply:
Done. And in the other file.